### ISSUE TITLE
New beta client doesn't remember login details.

### ISSUE THREAD

**Issue by galaxy119** [ValveSoftware/steam-for-linux#6527]
#### Your system information

* Steam client version (build number or date): Built 9/23/2019 @ 16:07:56
* Distribution (e.g. Ubuntu): Fedora 30
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Everytime I start steam, even with the "Remember Me" box checked, I am prompted for both a username and password.
#### Steps for reproducing this issue:

1. Enter login details, with the "Remember Me" box checked.
2. After logging in, exit steam.
3. Open steam.

Expected result: 
Steam remembers my account info and logs me in automatically.

**Comment by kisak-valve** [comment]
Hello @galaxy119, this is most likely a duplicate of #5030 / #5035. Closing in favor of the existing issue reports.

**Comment by galaxy119** [comment]
That issue is over 2 years old, and this has only started occuring for me since the last beta update (and reverting to the non-beta client solves the issue for me), so I'm not sure if this is really the same issue.

**Comment by kisak-valve** [comment]
Fair enough, if anyone else happens across this issue report, please only comment here if you are also not affected when using the pre-Library beta client, otherwise use #5030.

**Comment by galaxy119** [comment]
I've been doing a little digging around and think I've found the problem.

While many of the others in the previously linked issues seem to have this or a similar issue all the time, the latest 'workaround' they posted does work for me aswell.

What's happening is, when the steam client starts up, it's copying the ~/.steam/registry.vdf file to registry.vdf.old, then deleting the current one. It then re-makes it because it's not there anymore.
Disallowing create/delete files permissions in that folder solves the problem, as steam can no longer delete the currenty registry and re-make it.

**Comment by Luticus** [comment]
Just thought I'd throw my 2cents in on this because I've been dealing with this issue for, god, maybe years now and it's been driving me crazy.

I thought it was  a steam bug that would eventually get fixed so I kind of just left it alone and never bothered fixing it. Anyway I happened to notice it doesn't happen on a new laptop I got so I figured I'd take a second look.

@galaxy119 seems to be right about the registry.vdf.old being the culprit. However I couldn't think of a good clean way to prevent file deletion without messing with the file ownership and setting a sticky bit on the directory above, or making the file immutable via chattr +i, or by setting the permissions to remove write. None of those sat well with me as I thought they might mess up other things with steam, possible break updating, or have some side effect or unintended consequence. Fortunately I notices that the registry.vdf.old file had a file size of '0' and that to me looked... not right. I used cat to view the file's content and sure enough, it was blank. So I deleted the file and simply started steam. This seems to have resolved the issue for me.

You can verify this by watching the files change as you launch steam. Use the watch -n 1 "registry.*" and launch steam. You can see it create a file registry.vdf000000000 where the 0's represent some random number, in seems.  That's the behavior that was happening when it WORKED. So that's what you want. For some reason the .old file was preventing that... some how.

Also, in my case I symlinked a steam directory to .steam in my home to create a non hidden version for easy access. Not sure if it helped or not but I used unlink steam in my home directory to remove that as well.

I hope this advice helps anyone else facing this issue as it was driving me crazy!

**Comment by galaxy119** [comment]
For me, after deleting the vdf.old file, the next time I start steam, it just re-creates it, again with 0 bytes, and gives me the all too familiar login screen again.

**Comment by Luticus** [comment]
@galaxy119 , I had that issue too if I had a ~/steam folder or symlink. Try double checking that you only have a ~/.steam folder (hidden). If you have a symlink then use unlink to remove it, otherwise mv the folder from steam to .steam and see if that helps. If that's not your issue then there's obviously something else going on that's outside of what was causing it for me.
